transport.go

```diff
@@ -2,20 +2,56 @@ package transport
 
 import (
 	"context"
+	"io"
 	"net"
 	"time"
 
 	logging "github.com/ipfs/go-log"
+	smux "github.com/libp2p/go-stream-muxer"
 	ma "github.com/multiformats/go-multiaddr"
-	manet "github.com/multiformats/go-multiaddr-net"
 )
 
 var log = logging.Logger("transport")
 
 // Conn is an extension of the net.Conn interface that provides multiaddr
 // information, and an accessor for the transport used to create the conn
 type Conn interface {
-	manet.Conn
+	io.Closer
+
+	RemoteAddr() net.Addr
+	RemoteMultiaddr() ma.Multiaddr
+
+	LocalAddr() net.Addr
+	LocalMultiaddr() ma.Multiaddr
+
+	Transport() Transport
+}
+
+// A SingleStreamConn is a connection that provides a single channel between two endpoints
+// e.g. a TCP connection
+type SingleStreamConn interface {
+	Conn
+
+	io.Reader
+	io.Writer
+
+	SetDeadline(time.Time) error
+	SetReadDeadline(time.Time) error
+	SetWriteDeadline(time.Time) error
+}
+
+// A MultiStreamConn is a connection that supports transport-level stream multiplexing.
+// e.g. a QUIC connection
+// The MultiStreamConn combines the smux.Conn and the Conn interface
+// (unfortunately Go still doesn't allow duplicate interface methods...)
```

Comment:
We can still assert that this implements the correct interfaces:

```go
var _ Conn = (MultiStreamConn)(nil)
```

Reply:
Done.